File: frontend/src/types/article.ts
export interface Tag {
  id: number
  name: string
  slug: string
  articleCount?: number
}

export interface Author {
  id: number
  name: string
  avatar: string
}

export interface Article {
  id?: number
  title: string
  summary?: string
  content: string
  coverImage?: string
  authorId?: number
  viewCount?: number
  likeCount?: number
  commentCount?: number
  status?: number
  isTop?: number
  publishedAt?: string
  createdAt?: string
  updatedAt?: string
  author?: Author
  tags?: Tag[]
}

export interface ArticleListParams {
  page?: number
  pageSize?: number
  tag?: string
}

export interface ArticleListResponse {
  articles: Article[]
  total: number
}

File: backend/src/main/java/com/blog/entity/Article.java
package com.blog.entity;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class Article {
    private Integer id;
    private String title;
    private String summary;
    private String content;
    private String coverImage;
    private Integer authorId;
    private Integer viewCount;
    private Integer likeCount;
    private Integer commentCount;
    private Integer status;
    private Integer isTop;
    private LocalDateTime publishedAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    // 关联数据
    private User author;
    private List<Tag> tags;
}
